server/src/main/java/org/apache/iotdb/db/query/executor/fill/LinearFill.java

```diff
@@ -65,6 +68,30 @@ public LinearFill(TSDataType dataType, long queryTime, long beforeRange, long af
     this.afterRange = afterRange;
   }
 
+  public LinearFill() {}
+
+  public void convertRange(long startTime, long endTime) {
+    if (beforeRange % GroupByEngineDataSet.MS_TO_MONTH == 0) {
+      Calendar calendar = Calendar.getInstance();
```

Comment:
Please use the new date and time API defined in java8. See org.apache.iotdb.db.qp.utils.DatetimeUtils for examples.